packages/integration/src/azure/config.ts

```diff
@@ -33,25 +33,26 @@ export type AzureIntegrationConfig = {
   host: string;
 
   /**
-   * The authorization token to use for requests.
+   * The credentials to use for requests. If multiple credentials are specified the first one that matches the organization is used.
+   * If not organization matches the first credential without an organization is used.
    *
-   * If no token is specified, anonymous access is used.
+   * If no credentials are specified at all, anonymous access is used.
    */
-  token?: string;
+  credentials?: AzureIntegrationCredentialConfig[];
```

Comment:
We can't remove the `token` at this point, it'd be an immediate breaking change without deprecation period. If we feel it makes sense to remove in the future then we could deprecate it. I'm not entirely convinced we should do it, although in particular if we want to nudge people towards other options then I think it makes sense.

For now I suggest we generate a default credentials entry using the token if configured. It might also make sense to reject config that lists both a top-level token and credentials

Reply:
Done, I marked both `token` and `credential` as `@deprecated`. I did the scaffolder and catalog plugins for Azure to use the new `credentials` format.

Reply:
I also moved `getAzureRequestOptions` to `deprecated.ts` after removing all references to it from the code base. That way plugins that use this function will not break instantly.